tests/test_fits.py

```diff
@@ -128,6 +128,24 @@ def testUndefinedVector(self):
         metadata = self.writeAndRead(header)
         self.assertEqual(metadata.getArray("FOO"), [None, None])
 
+    def test_ticket_dm_36207(self):
+        # test whether moving to invalid HDU and then moving back throws or not
+        testfile = os.path.join(testPath, "data", "parent.fits")
+        fts = lsst.afw.fits.Fits(testfile, "r")
+
+        # parent.fits has 2 HDUs.
+        with self.assertRaises(lsst.afw.fits.FitsError):
+            fts.setHdu(5)
+
+        exc = None
+        try:
+            fts.setHdu(0)
```

Comment:
I think rather than proving no exception was raised, I'd rather prove we did succeed in moving to the first HDU by reading it and proving that we got a header from the right HDU.